bakery-js/src/epub/toc.tsx

```diff
@@ -176,6 +176,34 @@ export class TocFile extends BaseTocFile<
     // Add the epub:type="nav" attribute
     doc.findOne('//h:nav').attr('epub:type', 'toc')
 
+    // create extra elements when cover existing
+    if (this.parsed.coverFile) {
+      const body = doc.find('//h:body')[0]
+      const xmldoc = await this.readXml()
```

Comment:
I'm not sure this file needs to be parsed.

I think that something like this will work:

```tsx
// create extra elements when cover existing
if (this.parsed.coverFile) {
  const body = doc.findOne('//h:body')
  const source = getPos(doc.node)
  // Create the EPUB hidden nav landmarks for the cover and ToC
  const nav = doc.create('h:nav', {'epub:type': 'landmarks', 'hidden': 'hidden'}, [
    doc.create('h:ol', {}, [
      doc.create('h:li', {}, [
        doc.create('h:a', {'epub:type': 'cover', 'href': 'cover.xhtml'}, [ 'Cover' ], source)
      ], source),
      doc.create('h:li', {}, [
        doc.create('h:a', {'epub:type': 'toc', 'href': '#toc'}, [ 'Table of Contents' ], source)
      ], source)  
    ], source)
  ], source)
  body.node.insertBefore(nav.node, body.node.firstChild)
}
```


Or, this one with even less boilerplate:

```tsx
// create extra elements when cover existing
if (this.parsed.coverFile) {
  const body = doc.findOne('//h:body')
  // Create the EPUB hidden nav landmarks for the cover and ToC
  const nav = fromJSX(<h:nav epub:type='landmarks' hidden='hidden'>
    <h:ol>
      <h:li>
        <h:a epub:type='cover' href='cover.xhtml'>Cover</h:a>
      </h:li>
      <h:li>
        <h:a epub:type='toc' href='#toc'>Table of Contents</h:a>
      </h:li>
    </h:ol>
  </h:nav>)
  body.node.insertBefore(nav.node, body.node.firstChild)
}

// Add this to the bottom of `interface IntrinsicElements` in minidom.ts (around line 281)
'h:nav': { 'epub:type': 'landmarks'; hidden: 'hidden' }
'h:ol': {}
'h:li': {}
'h:a': { 'epub:type': 'cover' | 'toc'; href: string }
```

Reply:
oh, I tried to get it working with your minidom but did not know how. Was not aware the doc.create can be nested. Let me try if it works with either of those (will try fromJSX first)

Reply:
with fromJSX I've got problems the last time I've tried to use it on ol (but maybe I missed the namespace with `h:`)